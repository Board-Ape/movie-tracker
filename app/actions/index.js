export const moviesHasErrored = (bool) => {
  return {
    type: 'MOVIES_HAS_ERRORED',
    hasErrored: bool
  };
};

export const moviesIsLoading = (bool) => {
  return {
    type: 'MOVIES_IS_LOADING',
    isLoading: bool
  };
};

export const moviesFetchDataSuccess = (movies) => {

  return {
    type: 'MOVIES_FETCH_DATA_SUCCESS',
    movies
    };
};

export const fetchMovieList = (url) => {
  return (dispatch) => {
    dispatch(moviesIsLoading(true));
      fetch(url)
        .then((response) => {
          if (!response.ok) {
            throw Error(response.statusText);
          };
            dispatch(moviesIsLoading(false));
              return response;
            })
        .then((response) => response.json())
        .then(resp => {
          const movies = resp.results
          const newMovies = movies.map(movie => {
            return Object.assign({ isFavorite: false, movie_id: movie.id }, movie)
          })      
          return newMovies
        })
        .then((items) => dispatch(moviesFetchDataSuccess(items)))
        .catch(() => dispatch(moviesHasErrored(true)));
  };
};

export const fetchUserSigningIn = (email, password) => {
  return (dispatch) => {
    fetch('/api/users', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        email,
        password
      })
    })
    .then(results => results.json())
    .then(res => (dispatch(setUserToState(res.data))))
    .then(userData => (dispatch(fetchFavorites(userData.user.id))))
    .catch(res => alert("You're email and password do not match our records, please sign up!"))
  }
};

export const setUserToState = (user) => {
  return {
    type: 'SET_USER_TO_STATE',
    user
  }
}

export const AddUser = (name, password, email) => {
  return (dispatch) => {
    fetch('/api/users/new', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        name,
        password,
        email
      })
    })
    .then(results => results.json())
    .then(res => {
      if (!res.error) {
        (dispatch(setUserToState({
          id: res.id,
          name,
          password,
          email
        })))
      } else {
        alert("That Email as already been used.")
      }
    })
  }
}

export const fetchFavorites = (id) => {
  return (dispatch) => {
    fetch(`/api/users/${id}/favorites`)
    .then(response => response.json())
    .then( favResponse => {
        (favResponse.data.map(favorite=> {
        dispatch(toggleFavorite(favorite))
      }))
    })
    .then(res => console.log(res))
  }
}

export const AddFavorite = (movie, id) => {
  return (dispatch) => {
    fetch('/api/users/favorites/new', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({ movie_id: movie.id, user_id: id, title: movie.title, poster_path: movie.poster_path, release_date: movie.release_date, vote_average: movie.vote_average, overview: movie.overview })})
    .then(result => result.json())
    .then(response => dispatch(updateObj(movie.movie_id, response.id)))
  }
}

export const updateObj = (movie_id, favId) => {
    console.log('update: ', movie_id, favId)
  return {
    type: 'UPDATE_OBJ',
    movie_id,
    favId
  }
}

export const removeFavorite = (userId, movieId) => {
  console.log(userId, movieId)
  return (dispatch) => {
    fetch(`/api/users/${userId}/favorites/${movieId}`, {
      method: 'DELETE',
      headers: {
        'Content-type': 'application/json'
      },
      body: JSON.stringify({ user_id: userId, movie_id: movieId})})
      .then(result => result.json())
      .then(response => console.log(response))
  }
}

export const toggleFavorite = (movie) => {
  return {
    type: 'TOGGLE_FAVORITE',
    movie
  }
}

export const userSignOut = () => {
  return {
    type: 'USER_SIGNOUT'
  }
}

export const showFavorites = (bool) => {
  return {
    type: 'SHOW_FAVORITES',
    shouldShowFavorites: bool
  };
};
