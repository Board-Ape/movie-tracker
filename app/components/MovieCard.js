import React from 'react';

const MovieCard = () => {
  return (
    <div>
      Movie!
    </div>
  )
}

export default MovieCard;
