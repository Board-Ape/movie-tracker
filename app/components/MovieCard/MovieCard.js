import React from 'react';

const MovieCard = (props) => {
  return (
    <div className='movie-card'>
      <h1>{props.title}</h1>
      <img className='movie-poster'
           src={`https://image.tmdb.org/t/p/w500${props.poster}`}

            />
    </div>
  )
}
export default MovieCard;
