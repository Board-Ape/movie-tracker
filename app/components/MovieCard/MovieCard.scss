.movie-card {
  display: grid;
}

.movie-poster {
  width: 300px;
  justify-self: stretch;
}
