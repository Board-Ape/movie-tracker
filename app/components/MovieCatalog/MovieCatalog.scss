.home-page {
  display: grid;
}

.movie-cards {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  margin: auto;
  padding: auto;
}
