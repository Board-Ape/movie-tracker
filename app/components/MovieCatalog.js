import React, { Component } from 'react';
import { connect } from 'react-redux';
import MovieCard from './MovieCard';
import fetchMovieList from '../actions';
const key = require('../../utils/key');

class MovieCatalog extends Component {
  constructor() {
    super();
  }

  componentDidMount() {
    fetchMovieList(`https://api.themoviedb.org/3/movie/now_playing?api_key=${key}`)
  }

  render() {
    return (
      <div>
      MOVIES
      </div>
    )
  }
}

const mapStateToProps = (state) => {
    return {
        movies: state.movies,
        hasErrored: state.itemsHasErrored,
        isLoading: state.moviesIsLoading
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        fetchMovieList: (url) => dispatch(itemsFetchData(url))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(MovieCatalog);
