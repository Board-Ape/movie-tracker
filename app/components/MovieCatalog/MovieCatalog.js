import React, { Component } from 'react';
import { connect } from 'react-redux';
import MovieCard from '../MovieCard/MovieCard';
import Header from '../Header/Header';
import { fetchMovieList, setUserToState, shouldShowFavorites } from '../../actions';
const key = require('../../../utils/key');

class MovieCatalog extends Component {
  constructor() {
    super();
  }

  retrieveLocalStorage() {
    if (localStorage.getItem('user')) {
      this.props.handleSignInSuccess(JSON.parse(localStorage.getItem('user')))
    }
  }

  componentDidMount() {
    this.props.fetchMovieList(`https://api.themoviedb.org/3/movie/now_playing?api_key=${key}`)
    // this.retrieveLocalStorage();
  }

  render() {
    // console.log('catalogFav: ', this.props.shouldShowFavorites)
    // console.log(this.props.favorites)
    const moviesToRender = this.props.shouldShowFavorites === true ? this.props.favorites : this.props.movies
    // const movieArray = moviesToRender;
    const movieCardsArray = moviesToRender.map( (movie) => {
      console.log(movie)
      return (<MovieCard key={movie.id} movie={ movie }/>)
    })
    return (
      <div className='home-page'>
        <div className='movie-cards'>
          {movieCardsArray}
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    movies: state.movies,
    favorites: state.favorites,
    hasErrored: state.moviesHasErrored,
    isLoading: state.moviesIsLoading,
    user: state.user,
    shouldShowFavorites: state.shouldShowFavorites
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchMovieList: (url) => {
      dispatch(fetchMovieList(url))
    },
    handleSignInSuccess: (data) => {
      dispatch(setUserToState(data))
    },
    showFavorites: (bool) => {
      dispatch(showFavorites(bool))
    }
  }
};


export default connect(mapStateToProps, mapDispatchToProps)(MovieCatalog);
