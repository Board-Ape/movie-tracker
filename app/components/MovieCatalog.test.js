import React from 'react';
import { shallow, mount, configure } from 'enzyme'
import MovieCatalog from './MovieCatalog/MovieCatalog'
import MovieCard from './MovieCard/MovieCard';
import movie1 from '../../utils/mockData';
import Adapter from 'enzyme-adapter-react-16';
import * as actions from '../actions/index'

configure({ adapter: new Adapter() })

describe('MovieCatalog', () => {
  let wrapper;
  const mockMovie = movie1

  beforeEach( () => {
    wrapper = shallow(<MovieCatalog />);
  });

  it.skip('should create an instance of a MovieCatalog', () => {

    expect(wrapper).toMatchSnapshot();
  })
});
