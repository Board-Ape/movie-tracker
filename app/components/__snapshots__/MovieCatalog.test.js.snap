// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCatalog should create an instance of a MovieCatalog 1`] = `undefined`;
